### PR TITLE
Support go to line 1

### LINKED ISSUE #150: Support goto line 1
Basically fix this TODO: https://github.com/helix-editor/helix/blob/8c2fa12ffc006de787992b4022263be033ac8200/helix-term/src/commands.rs#L1286-L1287

As suggested by @pickfire in https://github.com/helix-editor/helix/pull/119#discussion_r646076087_, we could use `count: Option<NonZeroUsize>` instead of `count: usize` in `Context` and distinguish between explicitly set or not.

### PR DESCRIPTION
**Pull request by duduainankai**
Close #150

### CODE REVIEW

File: helix-term/src/ui/editor.rs [hunk 1]
```diff
@@ -532,7 +532,11 @@ impl EditorView {
             }
             _ => {
                 // set the count
-                cxt.count = cxt.editor.count.take().unwrap_or(1);
+                if let Some(count) = cxt.editor.count.take() {
+                    if let Some(count) = std::num::NonZeroUsize::new(count) {
+                        cxt.count = Some(count);
+                    }
+                }
```
Comment (CBenoit) [inline helix-term/src/ui/editor.rs:539]: I suggest to also change Editor's `count` to be a `NonZeroUsize` and simply do

```suggestion
                cxt.count = cxt.editor.count.take()
```

**Review by CBenoit — COMMENTED** [review]
Thank you!

File: helix-term/src/commands.rs [hunk 2]
```diff
@@ -95,6 +95,11 @@ impl<'a> Context<'a> {
         });
         self.callbacks.push(callback);
     }
+
+    #[inline]
+    pub fn get_count_or_one(&self) -> usize {
```
Comment (archseer) [inline helix-term/src/commands.rs:100]: maybe just call this `count()` and name the variable `_count`? Or `get_count()`

Comment (pickfire) [reply helix-term/src/commands.rs:100]: I agree on `count()`.

Comment (duduainankai) [reply helix-term/src/commands.rs]: updated.

**Comment by archseer** [comment]
Thanks!

**Pull request merged**